### DISCUSSION TITLE
Simple Setup script for Share Point libraries/Multiple Onedrive Accounts

### DISCUSSION THREAD

**Discussion by ghost** [abraunegg/onedrive#2055] [Show and tell]
Made a simple script for automatically creating and configuring Share Point libraries and/or other onedrive accounts apart from the main/default onedrive account. currently known to work on debian 11 with the sync directory name NOT containing spaces (may work with spaces, just haven't tried it).

[link to scripts](https://gist.github.com/Killerskyman/20834619c6d77ab88c37d38b9f354743)

script basically just creates the required config and sync directory based on user input. then it goes through the onedrive authentication process. then it creates a user level systemd service and enables and starts it. There is also a script included for deleting everything that the setup script makes.

The script should NOT be run with sudo, it does everything with normal user level permissions. For input the script will prompt you after you start it.
'Usage':
1. start script
2. input the name you want to have for the sync directory (example: "test")
3. enter the name of the sharepoint drive
4. enter the id of the drive (if you didn't guess the drive name correctly in step 3, just hit enter and it will loop back to step 3).
5. if no drive id was input, go to step 3. other wise the drive_id in the config will be set by the script.
6. at this point the script has everything it needs and will finish the config and make/start the systemd service.

Made this because I use teams a lot and all the channel folders in teams are technically share point libs and it was getting cumbersome to make the config and systemd service for automatic updating. hope some other people find this useful as well.

I might make updates to this in the future. one thing i definitely don't like with it right now is that it has the systemd service file is hard coded rather than say looking through the default onedrive.service file and finding the "ExecStart=..." line and editing that. just means that if onedrive updates the service file with new parameters this script may not work anymore.

**Comment by abraunegg** [comment]
@Killerskyman 
Thanks for contributing back to the community
